addons/stock/models/product.py

```diff
@@ -288,13 +288,14 @@ def _search_qty_available(self, operator, value):
         # to use the quants, not the stock moves. Therefore, we bypass the usual
         # '_search_product_quantity' method and call '_search_qty_available_new' instead. This
         # allows better performances.
+        search_domain = self._search_product_quantity(operator, value, 'qty_available')
```

Comment:
This change completely defeats the point of using `_search_qty_available_new` and avoiding using `_search_product_quantity` when it's not necessary (please read the code comment next time and try to understand what the code is doing before making your change). Please add new UserError checks to the `_search_qty_available_new` method instead